diff -Naur a/fs/open.c b/fs/open.c
--- a/fs/open.c	2025-07-23 16:10:34.546806069 +0200
+++ b/fs/open.c	2025-07-23 16:12:25.779439029 +0200
@@ -561,7 +561,7 @@
 SYSCALL_DEFINE3(faccessat, int, dfd, const char __user *, filename, int, mode)
 {
 #if defined(CONFIG_KSU) && !defined(CONFIG_KSU_KPROBES_HOOK)
-	ksu_handle_faccessat(&dfd, &filename, &mode, NULL);
+	ksu_handle_faccessat(&dfd, &filename, &mode, 0);
 #endif
 	return do_faccessat(dfd, filename, mode, 0);
 }
