diff -Naur a/kernel/selinux/selinux.c b/kernel/selinux/selinux.c
--- a/kernel/selinux/selinux.c	2025-07-23 16:34:05.086292010 +0200
+++ b/kernel/selinux/selinux.c	2025-07-23 17:02:38.450579439 +0200
@@ -60,7 +60,7 @@
 }*/
 }
 
-void setenforce(bool enforce)
+void ksu_setenforce(bool enforce)
 {
 #ifdef CONFIG_SECURITY_SELINUX_DEVELOP
 #ifdef KSU_COMPAT_USE_SELINUX_STATE
@@ -71,7 +71,7 @@
 #endif
 }
 
-bool getenforce()
+bool ksu_getenforce()
 {
 #ifdef CONFIG_SECURITY_SELINUX_DISABLE
 #ifdef KSU_COMPAT_USE_SELINUX_STATE
